httpx/_models.py

```diff
@@ -1243,13 +1243,12 @@ def iter_raw(self, chunk_size: int = None) -> typing.Iterator[bytes]:
 
     def close(self) -> None:
         """
-        Close the response and release the connection.
+        Mark the response as being closed. This will ensure that the response
+        body is no longer readable.
+
         Automatically called if the response body is read to completion.
         """
-        if not self.is_closed:
-            self.is_closed = True
-            if self._on_close is not None:
-                self._on_close(self)
+        self.is_closed = True
```

Comment:
It seems we still have `on_close: Callable = None` as part of the Response constructor for now, right? (Probably an oversight given the PR description.)